frontend/compositions/dashboard-header/desktop-nav.tsx

```diff
@@ -1,29 +1,32 @@
 import * as React from "react"
 import styles from "./dashboard-header.module.css"
+import Link from "next/link"
+import { useRouter } from "next/router"
 
 export default function DesktopNav() {
-  const [selected, setSelected] = React.useState("Search")
+  const router = useRouter()
 
-  const handleNavChange = (e: React.MouseEvent<Element>): void => {
-    const value = (e.target as HTMLElement).innerText
-    setSelected(value)
-  }
+  const menu = [
+    { title: "Search", path: "/dashboard" },
```

Comment:
I think we should change the search route to "/search" for clarity. /search and /profile will share the header component. We will also need to update existing references to "/dashboard" to "/search". This could be done in a separate PR.

Reply:
@alexjball Hi Alex, thanks so much for the code review. I _think_ I’m understanding the main reason behind the suggestion to move Dashboard Header into <Layout>, but wanted to check in before I start shifting things around. 

My current understanding:
- Next’s Layout component is useful for holding components that are reused across every page, like a header, nav bar, or footer. This prevents unnecessary re-rendering of repeated components.
- We’ve setup a shared Layout component, but it’s currently not holding any components. 
- We’ve wrapped every page in the above Layout component. Since we don’t have anything in Layout currently, it’s behaving more like a wrapper than an actual Layout component.

Proposed changes:
- Move the Dashboard Header component into the Layout component. 
- Wrap MyApp(pages/_app.js) with the Layout component, and remove the Layout component from every page.
- Now, whenever a route changes, Next re-renders MyApp with the new page component (but avoids re-rendering the header, since it’s part of the set Layout now)

My question: 
 - Testing the proposed changes above locally, every page will now display the Dashboard Header at the top. However we don’t want to display the header for some pages (log-in, passport). Would a conditional check/display for the header, based on the current pathname, in Layout be enough to address this? I’m not sure if I’m overthinking this. 

Please let me know if I'm off-track in any of the above, or can provide any more context/details. 


- [ ]

Reply:
Hey @nancyyang94 , great point, you are correct! putting the layout in _app would give every page the header, which we don't want to do for auth screens. the conditional approach could work. Overall I wonder if avoiding the rerender is worth the effort here, since it also makes things more complicated. I would lean toward keeping things as-is and revisiting if it becomes an issue.

Reply:
sounds good - will plan to keep things as is, and return to this if needed!